async_service/trio.py

```diff
@@ -144,8 +113,13 @@ def is_started(self) -> bool:
 
     @property
     def is_cancelled(self) -> bool:
-        return self._cancelled.is_set()
+        if self._task_nursery is None:
+            return False
+        cancel_scope = self._task_nursery.cancel_scope
+        return cancel_scope.cancel_called or cancel_scope.cancelled_caught
 
+    # XXX: ISTM this is unnecessary in TrioManager as nothing happens between the stopping and
+    # finished states...
```

Comment:
cc @ralexstokes I think this is where alex's request for a reliable way to run cleanup code would come in.  In the current model there is no way to reliably run a block of code that will run **after** all of the service code has stopped but *before* an external observer would see that the service is in the `finished` state.

Reply:
Should I just leave this comment in for now, then?

Reply:
I don't think so.  It's correct that there's no actual difference for the trio service but I don't see a need to comment about it since I'm pretty sure we're going to add the described functionality above.